src/main/scala/pro/civitaspo/digdag/plugin/ecs_task/register/EcsTaskRegisterOperator.scala

```diff
@@ -1,409 +1,222 @@
     extends AbstractEcsTaskOperator(operatorName, context, systemConfig, templateEngine) {
 
-  protected val config: Config = params.getNested("_command")
+  protected val config: DigdagConfig = DigdagConfig(params).getNested("_command")
 
-  protected def buildRegisterTaskDefinitionRequest(c: Config): RegisterTaskDefinitionRequest = {
+  protected def buildRegisterTaskDefinitionRequest(c: DigdagConfig): RegisterTaskDefinitionRequest = {
     val req: RegisterTaskDefinitionRequest = new RegisterTaskDefinitionRequest()
 
-    val containerDefinitions: Seq[ContainerDefinition] =
-      c.parseList("container_definitions", classOf[Config]).asScala.map(configureContainerDefinition).map(_.get)
-    val cpu: Optional[String] = c.getOptional("cpu", classOf[String])
-    val executionRoleArn: Optional[String] = c.getOptional("execution_role_arn", classOf[String])
-    val family: String = c.get("family", classOf[String])
-    val ipcMode: Optional[String] = c.getOptional("ipc_mode", classOf[String])
-    val memory: Optional[String] = c.getOptional("memory", classOf[String])
-    val networkMode: Optional[String] = c.getOptional("network_mode", classOf[String])
-    val pidMode: Optional[String] = c.getOptional("pid_mode", classOf[String])
-
-    val placementConstraints: Seq[TaskDefinitionPlacementConstraint] =
-      c.parseListOrGetEmpty("placement_constraints", classOf[Config]).asScala.map(configureTaskDefinitionPlacementConstraint).map(_.get)
-    val requiresCompatibilities: Seq[String] = c.parseListOrGetEmpty("requires_compatibilities", classOf[String]).asScala // Valid Values: EC2 | FARGATE
-    val tags: Seq[Tag] = c.getMapOrEmpty("tags", classOf[String], classOf[String]).asScala.map(t => new Tag().withKey(t._1).withValue(t._2)).toSeq
-    val taskRoleArn: Optional[String] = c.getOptional("task_role_arn", classOf[String])
-    val volumes: Seq[Volume] = c.parseListOrGetEmpty("volumes", classOf[Config]).asScala.map(configureVolume).map(_.get)
-
-    req.setContainerDefinitions(containerDefinitions.asJava)
-    if (cpu.isPresent) req.setCpu(cpu.get)
-    if (executionRoleArn.isPresent) req.setExecutionRoleArn(executionRoleArn.get)
-    req.setFamily(family)
-    if (ipcMode.isPresent) throw new UnsupportedOperationException("Currently aws-java-sdk does not support ipc_mode.")
-    if (memory.isPresent) req.setMemory(memory.get)
-    if (networkMode.isPresent) req.setNetworkMode(networkMode.get)
-    if (pidMode.isPresent) throw new UnsupportedOperationException("Currently aws-java-sdk does not support pid_mode.")
-    if (placementConstraints.nonEmpty) req.setPlacementConstraints(placementConstraints.asJava)
-    if (requiresCompatibilities.nonEmpty) req.setRequiresCompatibilities(requiresCompatibilities.asJava)
-    if (tags.nonEmpty) req.setTags(tags.asJava)
-    if (taskRoleArn.isPresent) req.setTaskRoleArn(taskRoleArn.get)
-    if (volumes.nonEmpty) req.setVolumes(volumes.asJava)
+    req.setFamily(c.get("family"))
```

Comment:
could you follow the document order? See https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_RegisterTaskDefinition.html .